lib/rubocop/rspec/top_level_group.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module RSpec
+    # Helper methods for top level example group cops
+    module TopLevelGroup
+      extend RuboCop::NodePattern::Macros
+      include RuboCop::RSpec::Language
+
+      def_node_matcher :example_or_shared_group?,
+                       (ExampleGroups::ALL + SharedGroups::ALL).block_pattern
+
+      def on_block(node)
+        return unless respond_to?(:on_top_level_group)
```

Comment:
An afterthought.

I understand that this guard statement is on the very top to avoid a quite expensive check done in `top_level_group?`.

But is it really necessary? If a cop includes `TopLevelGroup`, that means it has a `on_top_level_group` method?
Or am I missing something?

Reply:
TBH I just copied this approach from `TopLevelDescribe` module. I don't see any benefit in having such condition here.